keps/sig-cli/kubectl-apply-prune-beta.md

```diff
@@ -0,0 +1,250 @@
+---
+title: Kubectl Apply Prune Enhancement
+authors:
+  - "@Liujingfang1"
+owning-sig: sig-cli
+participating-sigs:
+  - sig-cli
+reviewers:
+  - "@pwittrock"
+  - "@seans3"
+  - "@soltysh"
+approvers:
+  - "@pwittrock"
+  - "@seans3"
+  - "@soltysh"
+editor: "Liujingfang1"
+creation-date: 2019-02-04
+last-updated: 2019-02-04
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - n/a
+superseded-by:
+  - n/a
+---
+
+# Kubectl Apply Prune Enhancement
+
+## Table of Contents
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [Challenges](#challenges)
+    * [Limitation of Alpha Prune](#limitation-of-alpha-prune)
+    * [Apply Config Object](#apply-config-object)
+    * [Child Objects](#child-objects)
+    * [How It Works](#how-it-works)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks](#drawbacks)
+* [Alternatives](#alternatives)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+Link to tracking issue: [kubernetes/kubectl#572](https://github.com/kubernetes/kubectl/issues/572)
+
+## Summary
+
+`kubectl apply --prune` has been in Alpha for more than a year. This design proposes a better approach of handling `--prune` and move it to Beta. This feature targets 1.15 release.
```

Comment:
Should this move it to beta or a second alpha iteration? If the first pass at doing this hasn't work and no outside experiment as proven this out... is it ready for beta? This is just me being conservative.

Reply:
From the SIG-CLI meeting discussion, I think we'd better make it a second alpha iteration.